contracts/DeployDebtToken.sol

```diff
@@ -0,0 +1,52 @@
+import './DebtToken.sol';
+pragma solidity ^0.4.15;
+contract DeployDebtToken is Ownable{
+
+    address public owner;
+    address public dayTokenAddress;
+    uint public dayTokenFees; //DAY tokens to be paid for deploying custom DAY contract
+    ERC20 dayToken;
+
+    event FeeUpdated(uint _fee, uint _time);
+    event DebtTokenCreated(address  _creator, address _debtTokenAddress, uint256 _time);
+
+    function DeployDebtToken(address _dayTokenAddress, uint _dayTokenFees){
+        dayTokenAddress = _dayTokenAddress;
+        dayTokenFees = _dayTokenFees;
+        owner = msg.sender;
+        dayToken = ERC20(dayTokenAddress);
+    }
+
+    function updateDayTokenFees(uint _dayTokenFees) onlyOwner public {
+        dayTokenFees = _dayTokenFees;
+        FeeUpdated(dayTokenFees, now);
+    }
+
+    function createDebtToken(string _tokenName,
+        string _tokenSymbol,
+        uint256 _initialAmount,
+        uint256 _exchangeRate,
+        uint256 _decimalUnits,
+        uint256 _dayLength,
+        uint256 _loanTerm,
+        uint256 _loanCycle,
+        uint256 _interestRate,
+        address _debtOwner){
+
+        address user = msg.sender;
+
+        if(dayToken.transferFrom(user, this, dayTokenFees)){
+            DebtToken newDebtToken = new DebtToken(_tokenName, _tokenSymbol, _initialAmount, _exchangeRate,
```

Comment:
When calling new DebtToken from DeployDebtToken the msg.sender seen by DebtToken will be the address of DeployDebtToken contract.

To fix this we shall modify the contract so it accepts 'borrower' as an input.

Reply:
Great Observation, we would create a different `DeployableDebtToken` with that feature, so we can separate development process of the two

Reply:
I really like the way `DayToken` has resolved it. In case of debt contract, I think it has too much of overhead. If we "rename" owner to "borrower" there is no need for 3rd party like owner.

My proposal is to
* add explicit argument to ctor `borrower`
* remove `owner` variable and semantics and replace with `borrower` (remove ownable inheritance)